docs/reference/search/retriever.asciidoc

```diff
@@ -201,6 +204,70 @@ GET /index/_search
 ----
 // NOTCONSOLE
 
+[[text-similarity-reranker-retriever]]
+==== Text Similarity Re-ranker Retriever
+
+The `text_similarity_reranker` is a type of retriever that enhances search results by re-ranking documents based on semantic similarity to a specified inference text, using a machine learning model.
+
+===== Prerequisites
+
+To use `text_similarity_reranker` you must first set up a `rerank` task using the <<put-inference-api, Create {infer} API>>.
+The `rerank` task should be set up with a machine learning model that can compute text similarity.
+Currently you can integrate directly with the Cohere Rerank endpoint using the <<infer-service-cohere,`cohere-rerank`>> task, or upload a model to {es} <<inference-example-eland,using Eland>>.
+
+===== Parameters
+
+`field`::
+(Required, `string`)
++
+The document field to be used for text similarity comparisons. This field should contain the text that will be evaluated against the `inferenceText`.
+
+`inference_id`::
+(Required, `string`)
++
+Unique identifier of the inference endpoint created using the {infer} API.
+
+`inference_text`::
+(Required, `string`)
++
+The text snippet used as the basis for similarity comparison.
+
+`rank_window_size`::
+(Optional, `int`)
++
+The number of top documents to consider in the re-ranking process. Defaults to `10`.
+
+`min_score`::
+(Optional, `float`)
++
+Sets a minimum threshold score for including documents in the re-ranked results. Documents with similarity scores below this threshold will be excluded. Note that score calculations vary depending on the model used.
+
+===== Restrictions
+
+A text similarity re-ranker retriever is a compound retriever. Child retrievers may not use elements that are restricted by having a compound retriever as part of the retriever tree.
+
+===== Example
+
+[source,js]
+----
+GET /index/_search
+{
+    "retriever": {
+        "text_similarity_reranker": {
+            "retriever": {
+                "standard": { ... }
+            }
+        },
+        "field": "text",
+        "inference_id": "my-cohere-rerank-model-v3",
```

Comment:
Nit: sorry, I left the `-v3` in due to a copy-paste because I used the Cohere Reranker English v3 model. To the reader it could be distracting, so let's remove it.

```suggestion
        "inference_id": "my-cohere-rerank-model",
```